Implemented Stories 2 and 3 (and 1)

TYKUHN2 opened the pull request:
These were similar enough I accidentally did both.

Commit:
Implemented Stories 2 and 3

These were similar enough I accidentally did both.

decepsis reviewed (approved):
This code works as expected. Now we just need to finish the stories 1, 4, and 5.

Commit:
Implemented Story 1

TYKUHN2 commented:
Implemented Story 1

Commit:
Fixed Assumption Mismatch

Previous Deck assumed faceDown was irrelevant, and Player assumed all cards were faceDown. This fixes it so all discards are, when readded, faceDown.
